java-checks/src/main/java/org/sonar/java/checks/StaticMultithreadedUnsafeFieldsCheck.java

```diff
@@ -61,8 +68,20 @@ public void visitNode(Tree tree) {
   }
 
   private static boolean isForbiddenType(VariableTree variableTree) {
+    if (isForbiddenType(variableTree.type().symbolType())) {
+      return true;
+    }
     ExpressionTree initializer = variableTree.initializer();
-    return isForbiddenType(variableTree.type().symbolType()) || (initializer != null && !initializer.is(Tree.Kind.NULL_LITERAL) && isForbiddenType(initializer.symbolType()));
+    if (initializer == null) {
+      return false;
+    }
+    if (initializer.is(Tree.Kind.NULL_LITERAL)) {
+      return false;
```

Comment:
The two `if`s that return `false` can be combined into one:

    if (initializer == null || initializer.is(Tree.Kind.NULL_LITERAL)) {
      return false;
    }